metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -353,7 +353,7 @@ public ControllerResult<BrokerRegistrationReply> registerBroker(
         if (existing != null) {
             prevIncarnationId = existing.incarnationId();
             storedBrokerEpoch = existing.epoch();
-            if (heartbeatManager.hasValidSession(brokerId, existing.epoch())) {
+            if (heartbeatManager.hasValidSession(brokerId, existing.epoch()) && heartbeatManager.isBrokerActive(brokerId)) {
```

Comment:
could we just check if whether `existing` is fenced or in controlled shutdown?

Reply:
thanks 😅